dask_sql/mappings.py

```diff
@@ -128,10 +123,10 @@ def sql_to_python_value(sql_type: str, literal_value: Any) -> Any:
 
         return literal_value
 
-    elif sql_type.startswith("INTERVAL"):
+    elif sql_type == SqlTypeName.INTERVAL:
         # check for finer granular interval types, e.g., INTERVAL MONTH, INTERVAL YEAR
         try:
-            interval_type = sql_type.split()[1].lower()
+            interval_type = str(sql_type).split()[1].lower()
 
             if interval_type in {"year", "quarter", "month"}:
                 # if sql_type is INTERVAL YEAR, Calcite will covert to months
```

Comment:
Some of these earlier assumptions about calcite may not hold true for datafusion and will need updating as and when mappings are encountered

Reply:
Completely agree. Even though all the current tests are passing I am sure there are some missing edge cases. I was thinking about opening a issue just for that alone and was hoping that you might be able to help with that one since you have done a lot of work in the date/time area. Could I open an issue and assign it to you for a subsequent PR?